### PR TITLE
Split nocuda variant in separate default and rocm variants

### PR DESCRIPTION
**Pull request by traversaro**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

**Review by jan-janssen — APPROVED** [review]
Looks good to me - Thanks @traversaro

**Comment by traversaro** [comment]
It is the first time I try to do something like this, so I may be doing something wrong, if anyone wants to provide feedback it is welcome!

File: recipe/meta.yaml [hunk 1]
```diff
@@ -59,6 +63,12 @@ about:
   license: BSD-3-Clause
   license_file: COPYING
   summary: Portable Hardware Locality (hwloc)
+  description: |
+      This feedstock builds the hwloc library in the `libhwloc` conda package. Different variants of the package are built:
+      * `cpu` in the build string: Without CUDA or ROCM support, installed by default. Install specifically this version with `conda install libhwloc=*=*cuda*`.
```
Comment (isuruf) [inline recipe/meta.yaml:68]:
```suggestion
      * `cpu` in the build string: Without CUDA or ROCM support, installed by default. Install specifically this version with `conda install libhwloc=*=*cpu*`.
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -41,12 +45,12 @@ requirements:
...
     - libxml2
     - pthreads-win32            # [win]
-    - rocm-smi                  # [linux and cuda_compiler_version == "None" and build_platform == target_platform]
+    - rocm-smi                  # [linux and cuda_compiler_version == "None" and build_platform == target_platform and ROCM_COMPILATION == "enabled"]
   run:
     - libxml2
     - pthreads-win32            # [win]
     - __cuda                    # [cuda_compiler_version != "None"]
-    - rocm-smi                  # [linux and cuda_compiler_version == "None" and build_platform == target_platform]
+    - rocm-smi                  # [linux and cuda_compiler_version == "None" and build_platform == target_platform and ROCM_COMPILATION == "enabled"]
```
Comment (isuruf) [inline recipe/meta.yaml:53]:
```suggestion
    - rocm-smi                  # [linux and cuda_compiler_version == "None" and ROCM_COMPILATION == "enabled"]
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -20,13 +20,17 @@ build:
   number: {{ build_num }}
   skip: true  # [win and vc<14]
   skip: true  # [cuda_compiler_version != "None" and win]
+  skip: true  # [cuda_compiler_version != "None" and ROCM_COMPILATION == "enabled"]
+  skip: true  # [not linux64 and ROCM_COMPILATION == "enabled"]
   string: cuda{{ cuda_compiler_version | replace('.', '') }}_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version != "None"]
-  string: nocuda_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version == "None"]
+  string: rocm_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [ROCM_COMPILATION == "enabled"]
+  string: cpu_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version == "None" and ROCM_COMPILATION == "disabled"]
```
Comment (isuruf) [inline recipe/meta.yaml:27]:
```suggestion
  string: default_h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}  # [cuda_compiler_version == "None" and ROCM_COMPILATION == "disabled"]
```

**Comment by traversaro** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (traversaro) [reply recipe/meta.yaml]: Solved by https://github.com/conda-forge/libhwloc-feedstock/pull/66/commits/0d7bd50cbfbd9cbef4f3cbb34b67d9cb72cf1181 .

**Pull request merged**